### DISCUSSION TITLE
Run code after prerendered embedded Server component has been materialized  in the browser

### DISCUSSION THREAD

**Discussion by marinasundstrom** [dotnet/aspnetcore#49389] [Q&A] [answered]
This is about the Blazor architecture in the upcoming .NET 8:

How do I execute code (client JavaScript) after a Server-rendered components that has been pre-rendered as part of a SSR:ed page get materialized in the browser? 

``OnAfterRender`` doesn't handle this, because the component is rendered once as part of the containing page. There is no !firstRender executing after that, which makes sense. 

So shouldn't there be an event for when the component has been materialized in the browser? On which this kind of logic can be safely executed.

**Comment by javiercn** [comment, marked as answer]
When the components execute on the server they are gone by the time the HTML is transferred to the client. Executing a script in that case can be achieved by subscribing to "enhancedload" in the Blazor ecosystem or using a mutation observer to get notified when a given element appears on the page.

If we are talking interactive components, they will trigger an OnAfterRender the moment they become interactive and render in that mode for the first time.
